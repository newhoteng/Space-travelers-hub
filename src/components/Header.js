import { Outlet, NavLink } from 'react-router-dom';
import styles from '../styles/Header.module.css';
import planet from '../Assets/planet.png';

function Header() {
  const navLinkStyles = ({ isActive }) => ({
    textDecoration: isActive ? 'underline' : 'none',
  });

  return (
    <>
      <header className={styles.header}>
        <div>
          <img src={planet} alt="planet" />
          <h1>Space Travelers&apos; Hub</h1>
        </div>
        <nav>
          <ul className={styles.navigation}>
            <li><NavLink style={navLinkStyles} to="/">Rockets</NavLink></li>
            <li><NavLink style={navLinkStyles} to="/missions">Missions</NavLink></li>
            <li><NavLink style={navLinkStyles} to="/dragons">Dragons</NavLink></li>
            <li className={styles.item} />
            <li><NavLink style={navLinkStyles} to="/profile">My Profile</NavLink></li>
          </ul>
        </nav>
      </header>
      <Outlet />
    </>
  );
}

export default Header;
