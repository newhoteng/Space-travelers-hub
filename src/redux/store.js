import { configureStore } from '@reduxjs/toolkit';
import missionsReducer from './missions/missionsSlice';
import dragonReducer from './Dragons/DragonsSlice';
import rocketReducer from './Rockets/RocketsSlice';

export const store = configureStore({
  reducer: {
    missions: missionsReducer,
    dragon: dragonReducer,
    rocket: rocketReducer,
  },
});

export default store;
