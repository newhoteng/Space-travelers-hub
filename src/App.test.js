import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { store } from './redux/store';

// Component imports
import Header from './components/Header';
import Profile from './components/MyProfile';

// Snapshots for Header and Profile components
describe('App snapshots', () => {
  it('header renders correctly', () => {
    const header = renderer
      .create(
        <BrowserRouter>
          <Header page="/">Header</Header>
        </BrowserRouter>,
      )
      .toJSON();
    expect(header).toMatchSnapshot();
  });

  it('profile renders correctly', () => {
    const profile = renderer
      .create(
        <Provider store={store}>
          <Profile page="/profile">Profile Page</Profile>
        </Provider>,
      )
      .toJSON();
    expect(profile).toMatchSnapshot();
  });
});

// Using React Testing library to render Header and Profile omponents
describe('Render Components', () => {
  it('renders Header component', () => {
    render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );
  });

  it('renders Profile component', () => {
    render(
      <Provider store={store}>
        <Profile />
      </Provider>,
    );
  });
});
