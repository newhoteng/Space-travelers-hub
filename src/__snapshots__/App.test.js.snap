// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshots header renders correctly 1`] = `
<header
  className="header"
>
  <div>
    <img
      alt="planet"
      src="planet.png"
    />
    <h1>
      Space Travelers' Hub
    </h1>
  </div>
  <nav>
    <ul
      className="navigation"
    >
      <li>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "underline",
            }
          }
        >
          Rockets
        </a>
      </li>
      <li>
        <a
          className=""
          href="/missions"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
          Missions
        </a>
      </li>
      <li>
        <a
          className=""
          href="/dragons"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
          Dragons
        </a>
      </li>
      <li
        className="item"
      />
      <li>
        <a
          className=""
          href="/profile"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
          My Profile
        </a>
      </li>
    </ul>
  </nav>
</header>
`;

exports[`App snapshots profile renders correctly 1`] = `
<div
  className="tablesContainer"
>
  <div
    className="eachTable"
  >
    <p>
      My Missions
    </p>
    <table
      className="profilePage"
    />
  </div>
  <div
    className="eachTable"
  >
    <p>
      My Rockets
    </p>
    <table
      className="profilePage"
    />
  </div>
  <div
    className="eachTable"
  >
    <p>
      My Dragons
    </p>
    <table
      className="profilePage"
    />
  </div>
</div>
`;
